patches/minecraft/net/minecraft/item/ItemStack.java.patch

```diff
@@ -284,4 +284,18 @@
 +            return this.capabilities.areCompatible(other.capabilities);
 +        }
 +    }
++
++    public static boolean areItemStacksWithShareTagEqual(@Nullable ItemStack stackA, @Nullable ItemStack stackB)
++    {
++        if (stackA == null && stackB == null) return true;
++        if (stackA == null && stackB != null) return false;
++        if (stackA != null && stackB == null) return false;
```

Comment:
Or even nicer: `if (stackA == null || stackB == null) return stackA == stackB;` to save two lines